hw4/task_01.py

```diff
@@ -0,0 +1,89 @@
+"""
+Написать программу, которая скачивает изображения с заданных URL-адресов и сохраняет их на диск. Каждое изображение
+должно сохраняться в отдельном файле, название которого соответствует названию изображения в URL-адресе.
+Например, URL-адрес: https://example/images/image1.jpg -> файл на диске: image1.jpg
+— Программа должна использовать многопоточный, многопроцессорный и асинхронный подходы.
+— Программа должна иметь возможность задавать список URL-адресов через аргументы командной строки.
+— Программа должна выводить в консоль информацию о времени скачивания каждого изображения и общем времени выполнения программы.
+"""
+import threading
+from multiprocessing import Process
+import time
+import asyncio
+import aiohttp
+import requests
+
+
+urls = ['https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911755.png',
+        'https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911757.png',
+        'https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911758.png',
+        'https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911759.png',
+        'https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911760.png',
+        'https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911761.png',
+        "https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911762.png",
+        "https://cloud.yatco.com/ForSale/Vessel/Photo/379295/large_3911763.png"]
+
+
+def download(url, path_):
+    start_time = time.time()
+    response = requests.get(url)
+    filename = path_ + url.replace('https://cloud.yatco.com/ForSale/Vessel/Photo/379295/', '').replace('/', '')
```

Comment:
Делайте, код всегда масштабируемым.
```suggestion
    _, filename = os.path.join(path_, url.rsplit('/', 1)
```
или
```suggestion
    filename = os.path.join(path_, os.path.basename(url))
```